include/bitcoin/bitcoin/wallet/electrum.hpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2011-2017 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef LIBBITCOIN_WALLET_ELECTRUM_HPP
+#define LIBBITCOIN_WALLET_ELECTRUM_HPP
+
+#include <cstddef>
+#include <string>
+#include <vector>
+#include <bitcoin/bitcoin/compat.hpp>
+#include <bitcoin/bitcoin/define.hpp>
+#include <bitcoin/bitcoin/math/hash.hpp>
+#include <bitcoin/bitcoin/unicode/unicode.hpp>
+#include <bitcoin/bitcoin/utility/data.hpp>
+#include <bitcoin/bitcoin/utility/string.hpp>
+#include <bitcoin/bitcoin/wallet/dictionary.hpp>
+
+namespace libbitcoin {
+namespace wallet {
+namespace electrum {
+
+typedef boost::multiprecision::cpp_int cpp_int;
+
+/**
+ * Valid seed types.
+ */
+enum class seed_type : uint8_t
+{
+    standard = 1,
+    segwit = 2,
+    two_factor_authentication = 3
+};
```

Comment:
Avoid use of "type" in types. `bc::wallet::electrum::seed` should be sufficient. Avoid the "segwith" acronym, instead use "witness".

Reply:
Addressing this now